sw/device/lib/dif/dif_hmac.h

```diff
@@ -0,0 +1,138 @@
+ */
+void dif_hmac_init(const dif_hmac_config_t *config, dif_hmac_t *hmac_out);
+
+/**
+ * Attempts to send |len| bytes from the buffer pointed to by |data| to the
+ * device described by |hmac|. This function will send to the message FIFO until
+ * the FIFO fills up or |len| bytes have been sent.
+ *
+ * |data| *must* point to an allocated buffer of at least length |len|.
+ *
+ * @param hmac the HMAC device to send to.
+ * @param data a contiguous buffer to copy from.
+ * @param len the length of the buffer to copy from.
+ * @return the number of bytes actually sent.
+ */
+size_t dif_hmac_push_fifo(const dif_hmac_t *hmac, void *data, size_t len);
+
+/**
+ * Retrieves the number of bits in the loaded HMAC device.
+ *
+ * @param hmac the HMAC device to get the message length for.
+ * @return the number of bits in the HMAC message.
+ */
+uint64_t dif_hmac_get_msg_len(const dif_hmac_t *hmac);
+
+/**
+ * Attempts to run HMAC or SHA256 depending on the mode |hmac| was initialized
+ * in. Calls to this function always succeed and return without blocking. The
+ * caller can use |dif_hmac_check_state()| to check for errors and for the
+ * |DIF_HMAC_DONE| status before reading the digest with
+ * |di_hmac_read_digest()|.
+ *
+ * @param hmac the HMAC device to initiate the run on.
+ */
+void dif_hmac_run(const dif_hmac_t *hmac);
+
+/**
+ * Checks the HMAC device to see if it has finished a run triggered by
+ * |dif_hmac_run()|. A logical true return value indicates that the HMAC run has
+ * completed and it is now safe to read the digest with |hmac_read_digest()|.
+ *
+ * Calling this function has the side effect of clearing the hmac_done status,
+ * meaning that after this function returns true subsequent calls will return
+ * false until |dif_hmac_run()| is called again.
+ *
+ * @param hmac the HMAC device to check.
+ * @return local true if this is the first time this function has been called
+ * since HMAC completed, false otherwise.
+ */
+bool dif_hmac_is_done(const dif_hmac_t *hmac);
+
+/**
+ * Attempts to read the HMAC digest and store store the result in the buffer
+ * referenced by |digest_out|. |dif_hmac_state()| should before this
+ * function to ensure the digest is available to be read.
+ *
+ * |digest_out| must reference an allocated, contiguous, 32-byte buffer.
+ *
+ * @param hmac the HMAC device to read the digest from.
+ * @param digest_out a contiguous 32-byte buffer to store the digest.
+ * @return logical true if the digest was read successfully, false otherwise.
+ */
+void dif_hmac_read_digest(const dif_hmac_t *hmac,
+                          dif_hmac_digest_t *digest_out);
+
+/**
+ * Randomizes internal secret registers on the HMAC device. This includes the
+ * key, hash value, and internal state machine. The value of |entropy| will be
+ * used to "randomize" the internal state of the HMAC device. See the HMAC IP
+ * documentation for more information: hw/ip/hmac/doc.
+ *
+ * @param hmac the HMAC device to clobber state on.
```

Comment:
`@param hmac the HMAC device to clobber state on.` could you start the variable description with a capital, like `@param hmac The ...`.